cmd2/cmd2.py

```diff
@@ -1810,47 +1804,25 @@ def _run_cmdfinalization_hooks(self, stop: bool, statement: Optional[Statement])
         except Exception as ex:
             self.perror(ex)
 
-    def runcmds_plus_hooks(self, cmds: List[str]) -> bool:
-        """Convenience method to run multiple commands by onecmd_plus_hooks.
+    def runcmds_plus_hooks(self, cmds: List[Union[HistoryItem, str]]) -> bool:
+        """
+        Used when commands are being run in an automated fashion like text scripts or history replays.
+        The prompt and command line for each command will be printed if echo is True.
 
-        This method adds the given cmds to the command queue and processes the
-        queue until completion or an error causes it to abort. Scripts that are
-        loaded will have their commands added to the queue. Scripts may even
-        load other scripts recursively. This means, however, that you should not
-        use this method if there is a running cmdloop or some other event-loop.
-        This method is only intended to be used in "one-off" scenarios.
+        :param cmds: commands to run
+        :return: True if running of commands should stop
+        """
+        for line in cmds:
+            if isinstance(line, HistoryItem):
+                line = line.raw
 
-        NOTE: You may need this method even if you only have one command. If
-        that command is a load, then you will need this command to fully process
-        all the subsequent commands that are loaded from the script file. This
-        is an improvement over onecmd_plus_hooks, which expects to be used
-        inside of a command loop which does the processing of loaded commands.
+            if self.echo:
+                self.poutput('{}{}'.format(self.prompt, line))
 
-        Example: cmd_obj.runcmds_plus_hooks(['load myscript.txt'])
+            if self.onecmd_plus_hooks(line):
```

Comment:
Ok, so now we just run all of the commands in a loop instead of appending them to the cmdqueue.  That seems reasonable.  I'm fearful that there could be some really obscure edge case where this would behave very differently, but I can't think of any right now.